pkg/issuer/acme/dns/webhook/webhook.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2018 The Jetstack cert-manager contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"github.com/golang/glog"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/jetstack/cert-manager/pkg/issuer/acme/dns/util"
+)
+
+const (
+	maxBodyLength = 5 * 1024 * 1024
+
+	presentOperation operation = "present"
+	cleanupOperation operation = "cleanup"
+)
+
+type operation string
+
+type WebhookPayload struct {
```

Comment:
I think we may also want to include a `metav1.TypeMeta` field here, which will allow us to actually version this webhook payload and allow webhook implementors to build webhooks that work against multiple versions of cert-manager in case we ever change it.

I'm happy to leave this *out* of this initial implementation, so that we can get this merged sooner. But it'd be great if you can open an issue to track this, as it's essential for versioning!